### PR TITLE
New Event: Acquiring a member of a royal family/court

### PR DESCRIPTION
**Pull request by freecitiesbandit**
First time using Git. Not sure if I set up the event correctly either, but hopefully the writing itself is fine.

### CODE REVIEW

**Comment by freecitiesdev** [comment]
Good. Needs some edits. ¤ needs to appear before every monetary amount. More seriously, the PC needs less characterization. Try describing the PC's behavior less explicitly. Someone who's playing their PC as a hard driven, hard working type is going to look at the stuff about not wanting to wake up in the middle of the night and think it doesn't describe their PC well. You have to be very careful describing the PC, so that it'll fit as many players' PCs as possible.

Generally, phrases like "you rarely bother with things like this" are dangerous, since some players don't think of their PC that way. I'm removing them as I find them in old events.

**Comment by freecitiesbandit** [comment]
Good comments. I'll make some edits.

**Comment by freecitiesbandit** [comment]
I made some edits, hopefully that addresses most of the concerns raised.

**Comment by freecitiesdev** [comment]
OK, next question: has this event tested OK in game? Recruit event coding will create horribly bugged slaves if it's even a little off.

**Comment by freecitiesbandit** [comment]
Not yet. What's the best way to testfire an event?

**Comment by freecitiesdev** [comment]
Easy. Just add <<set $events.push("RE royal blood")>> without the condition
to randomNonindividualEvent.tw
<https://github.com/Free-Cities/Free-Cities/pull/425/files#diff-af49489ecb11075980856df93a3b4bde>.
That'll force it into the random event rotation every week. Then enable
cheat mode, which lets you select a random event. You should see "RE royal
blood" on that menu.

**Comment by freecitiesbandit** [comment]
Thanks for the tip. I noticed that the female courtiers, queen and princess could generate with dicks, so I corrected that. I also cleared up two macro errors. The slaves seem to generate cleanly, as far as I can tell,

File: src/uncategorized/REroyalblood.tw [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+<span id="result">
+<<link "Go back to sleep">>
+	<<replace "#result">>
+	You inform your personal assistant that you aren't planning to take any action. It's not every day that a monarchy collapses in the Old World, but it's also not particularly uncommon either. 
+	<</replace>>
+<</link>>
+<br><<link "Leverage your reputation and credits to aid the new arcology in exchange for a pretty princess">>
+	<<replace "#result">>
+	You seize a tablet and practically roll out of your bed, working furiously and using every feature of $assistantName to the utmost. It's not easy, but you call in a number of favors and grease a handful of palms, and soon a flight of VTOLs are landing in the new arcology laden with goods. When they take off again they have the princess aboard clad in chains, and make a direct course towards  your waiting penthouse.
```
Comment (fcpmanon) [inline src/uncategorized/REroyalblood.tw]: Double space on line 31 of REroyalblood.tw

File: src/uncategorized/REroyalblood.tw [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+		<<set $activeSlave.health = random(30,60)>>
+		<<set $activeSlave.behavioralFlaw = either("bitchy", "arrogant")>>
+		<<AddSlave $activeSlave>>
+		<</for>>
+	<br><br>
+<</replace>>
+<</link>> // It will cost about ¤25000 to enslave them.//
+<br><<link "Transfer a modest quantity of credits for the Queen herself">>
+	<<replace "#result">>
+	You take a tablet and transmit a communication request to the new arcology owner. It doesn't take long for him to accept but when his video feed connects on your screen, you find yourself staring at the pained expression of the Queen herself. It becomes abundantly clear that someone is mounting her from behind as her body shakes and spasms from the intensity of their thrusts. You note  that the Queen appears to be getting fucked on a extravagant, yet bloodstained, four-poster bed, likely the site of her former husband's demise. The mysterious individual unsurprisingly turns out to be the new arcology owner, revealing himself by shoving the Queen's face down against the stained bedsheets as he lowers his own face into frame. "How can I help you?"
```
Comment (fcpmanon) [inline src/uncategorized/REroyalblood.tw]: Also double space on line 153

File: src/uncategorized/REroyalblood.tw [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+<br><<link "Leverage your reputation and credits to aid the new arcology in exchange for a pretty princess">>
+	<<replace "#result">>
+	You seize a tablet and practically roll out of your bed, working furiously and using every feature of $assistantName to the utmost. It's not easy, but you call in a number of favors and grease a handful of palms, and soon a flight of VTOLs are landing in the new arcology laden with goods. When they take off again they have the princess aboard clad in chains, and make a direct course towards  your waiting penthouse.
+<br><br>
+	Eventually she arrives in your penthouse, the perfect image of a demure yet composed princess. Her clearly practiced facade of poise and grace fades under scrutiny, however. The slightest trembling of her balled up fists, the minute tremors that mar her immaculate posture, her inability to meet your eyes with her own, all signs that she is still a scared girl despite all her royal trappings. Nonetheless, though the princess's court training is unlikely to be very beneficial to her in her new life in the penthouse, it does stand in stark contrast to her more common slave peers.
+	<<set $cash -= 50000>>
+	<<set $rep += 2000>>
```
Comment (fcpmanon) [inline src/uncategorized/REroyalblood.tw]: Wrong sign on line 35: should be -= not +=

**Comment by fcpmanon** [comment]
Consider setting intelligenceImplant = 1 for the royals, as they are presumably educated.

**Comment by fcpmanon** [comment]
Also consider surrounding each of the action options with checks for sufficient cash and reputation and display // You lack the funds to do this // or // You lack the reputation to do this // in the corresponding else blocks for each action ... or similar, as appropriate.

**Comment by fcpmanon** [comment]
I'm not sure what the best practice is, but this seems like it should be a one-off event, or at least very rare.  As written, if rep is 20000, the event will never fire since 20000 - 10000 = 10000 is never > 10000.  Consider changing the upper bound of the randomization to a (much) larger number to adjust the probability.

**Comment by freecitiesbandit** [comment]
I made the edits you suggested, and after a lot of struggle I think I managed to wrap each action with a cash/reputation check. During my quick check in-game, the "lack cash" message seems to fire if the cash threshold is not met. 

As for the randomization, I took the code pretty much wholesale from this:

<<if $RegularParties == 1>>
	<<set $events.push("RE citizen hookup")>>
	<<if $rep-10000 > random(1,10000)>>
		<<set $events.push("RE female arcology owner")>>
	<</if>>
<</if>>

I'm fairly sure I've seen this event fire in game, so will the royal event not fire?

**Comment by fcpmanon** [comment]
You're right.  I failed at maths and read it backwards.  10000 > 9999, so that will fire almost all the time if rep is 20000 and quite often even with lower rep.

**Comment by freecitiesbandit** [comment]
No worries, I had to have a close look at it to figure it out myself. I think it might still fire too often at 12500, but that's something that can be ironed out.

**Comment by fcpmanon** [comment]
All seems correct now.

File: src/uncategorized/REroyalblood.tw [hunk 4]
```diff
@@ -0,0 +1,216 @@
+:: RE royal blood [nobr]
+
+<<set $nextButton = "Continue", $nextLink = "RIE Eligibility Check">>
+
+One unfortunate consequence of existing in a global world is that business opportunities continue to crop up unabated by the onset of night in one corner of the world or another, an inconsiderate phenomenon for arcology owners who happen to be asleep when an event of importance takes place outside the walls of their arcology. This particular evening you are awoken by an alert from $assistantName, unusually intense for this late hour. 
+<<if $assistant > 0>>
+	" <<if def $PC.customTitle>>$PC.customTitle<<elseif $PC.title != 0>>Master<<else>>Mistress<</if>>, there is a time sensitive enslavement opportunity available if you're interested." She pauses before continuing. "You're going to want to see this one <<if def $PC.customTitle>>$PC.customTitle<<elseif $PC.title != 0>>Master<<else>>Mistress<</if>>.
```
Comment (freecitiesbandit) [inline src/uncategorized/REroyalblood.tw]: The If check on PC title should be changed to <<Master>>. 

I also accidentally pushed an extra event I've been working on into the pull request and my attempt to remove it seems to have have also removed my ability to edit the files in this request.

Comment (Frodfurgurn1991) [inline src/uncategorized/REroyalblood.tw]: consider using `<<Master>>`, I think it's equivalent

Comment (freecitiesbandit) [inline src/uncategorized/REroyalblood.tw]: Yup. I found it while browsing some of the edits to my previous events. It's very helpful.

**Pull request merged**